### PR TITLE
api: Enhance presignedGetObject API to support request params.

### PR DESCRIPTION
**Pull request by harshavardhana**
This is needed in cases when certain types of responses are
expected for a GET on presigned requests.

Supports response header overrides for

```
 [ response-expires,
   response-content-type,
   response-cache-control,
   response-content-disposition ]
```

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -1386,11 +1389,38 @@ public String presignedGetObject(String bucketName, String objectName, Integer e
     String region = BucketRegionCache.INSTANCE.region(bucketName);
 
     Request request = createRequest(Method.GET, bucketName, objectName, region,
-                                    null, null, null, null, 0);
-    HttpUrl url = Signer.presignV4(request, region, accessKey, secretKey, expires);
+                                    null, reqParams, null, null, 0);
+    HttpUrl url = Signer.presignV4(request, region, accessKey, secretKey,
+                                   expires);
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: You could avoid indentation here as 150 chars long line is allowed in checkstyle.

Comment (harshavardhana) [inline api/src/main/java/io/minio/MinioClient.java]: i saw it is 120.

**Comment by harshavardhana** [comment]
> Please run `./gradlew runFunctionalTest'

I already ran @balamurugana a new test is added as well. are you seeing any issue?

**Comment by balamurugana** [comment]
LGTM :+1:

**Pull request merged**